cpp/utility.cc

```diff
@@ -87,4 +88,14 @@ std::shared_ptr<grpc::ServerCredentials> GetServerCredentials(
   assert(0);
 }
 
+std::unique_ptr<grpc::Server> StartAdminServer(const std::string& port) {
+  std::string server_address = "localhost" + port;
+  grpc::EnableDefaultHealthCheckService(true);
```

Comment:
Is C++'s health checking service shared within the process or separated by server objects?

Reply:
I don't see the `port` being passed to `grpc::EnableDefaultHealthCheckService()` so how does this code ensure the health checking is started on `port` (which is the value of `admin_port`)?

Reply:
The listening port(s) for a server is different from the list of services registered with it. What `grpc::EnableDefaultHealthCheckService()` does is, it registers a health check service by default to servers, which allows the health check service to be exposed on the ports that it serves on.

Reply:
Okay it seems to be a global setting that affects all the servers created after this call. In that case the AdminServer will have the health checking enabled. LG

Reply:
> Is C++'s health checking service shared within the process or separated by server objects?

Each server uses a different health check service but the implementation itself can be shared. This is the implementation for the default health check service  - https://github.com/grpc/grpc/blob/master/src/cpp/server/health/default_health_check_service.cc